core/src/main/java/org/apache/carbondata/core/util/DeleteLoadFolders.java

```diff
@@ -188,8 +188,11 @@ public static boolean deleteLoadFoldersFromFileSystem(
               LOGGER.info("Info: Deleted the load " + oneLoad.getLoadName());
             }
           } finally {
-            segmentLock.unlock();
-            LOGGER.info("Info: Segment lock on segment:" + oneLoad.getLoadName() + " is released");
+            if (segmentLock.isLocked()) {
```

Comment:
FYI:
Only unlock if we lock it successfully, otherwise this will cause failure in windows.